.github/scripts/metrics_checker.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+set -o pipefail
```

Comment:
Suggest using `set -euo pipefail` for all the reasons mentioned [here](https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/) (minus the `-x` which is good for debugging imho). In general, using these options will help us catch strange side-effects as things around this script potentially change.

Reply:
Initially I had set the following but decided to go without:

`set -u`

for when we don't have variables hold a diff they are empty, which will fail the script. The alternative would be to assign placeholder values but I think that also can be a source of bugs!

Reply:
I always use `set -u`. To deal with variables that might be empty you can do this: `${varname-}`. That tells bash, "it's ok for this value to be unset". That way you have to explicit allow something being unset, instead of it being implied everywhere.

Reply:
You can see that in the docs here: https://tldp.org/LDP/abs/html/parameter-substitution.html, the second example

Reply:
thanks for the pointers -- i added that over the env var passed to the script 💯 !